mobile/native/src/ln_dlc/node.rs

```diff
@@ -371,6 +392,16 @@ impl Node {
     }
 }
 
+/// Returns true if the contract is already expired. Errors if the contract is not a valid unixe
+/// timestamp
+fn is_expired(offer: &SubChannelOffer) -> Result<bool> {
+    let now = OffsetDateTime::now_utc();
+    let offer_expiry =
+        OffsetDateTime::from_unix_timestamp(offer.contract_info.get_closest_maturity_date() as i64)
+            .context("Could not convert maturity date into offset date time")?;
```

Comment:
🔧 I think we should also reject the subchannel offer if we can't figure out the expiry. We should introduce a separate `InvalidSubchannelOffer` variant for that e.g. `UndeterminedMaturityDate`.

Reply:
e0021f615cfd6650312220308a996dc46b30a946